src/jit/emit.cpp

```diff
@@ -7277,3 +7303,41 @@ const char* emitter::emitOffsetToLabel(unsigned offs)
 }
 
 #endif // DEBUG
+
+//------------------------------------------------------------------------
+// GetSavedSet: Returns the set of registers that live across the call.
+//
+// Arguments:
+//   methHnd - the method handler of the call.
+//
+// Return value:
+//   the saved set of registers.
+//
+regMaskTP emitter::GetSavedSet(CORINFO_METHOD_HANDLE methHnd)
+{
+    // Is it a helper with a special saved set?
+    bool isNoGCHelper = emitNoGChelper(methHnd);
+    if (isNoGCHelper)
+    {
+        CorInfoHelpFunc helpFunc = Compiler::eeGetHelperNum(methHnd);
+
+        // Get the set of registers that this call kills and remove it from the saved set.
+        regMaskTP savedSet = RBM_ALLINT & ~emitComp->compNoGCHelperCallKillSet(helpFunc);
```

Comment:
The strange thing here is that we return only integer registers as alive here, when for the normal call we also return some float registers as alive, does anybody know is it intentional or not?

Reply:
Currently only integer registers can contain GC refererences.

Reply:
Ok, than I think we can rename this function to "GetSavedGCRegsSet` (how to combine byrefs and gcrefs together?) and return `RBM_INT_CALLEE_SAVED` in the else case. Does it make sence?

Reply:
renamed to `emitGetSavedGCRegsSet`.

Reply:
Naming here is going to be tricky -- I think `saved` gives the wrong idea. But not sure I have anything better to offer...

maybe `emitGetPreservedAndUpdatedGCRegsSet` ?

Reply:
I may not fully understand what's going on here, but I'm assuming that any of the registers in this set are either saved (i.e. not modified) or they are modified and contain GC refs after the call. Based on the usage, it seems that the latter happens only if the register contained a GC ref coming in (because it appears that the caller won't *add* any GC regs based on the returned value). If my understanding is correct, it would be good to add a note to that effect in the header of this method. And I might suggest `emitGetGCRegsSavedOrModified`.

Reply:
I added a note and renamed the function to `emitGetGCRegsSavedOrModified`. It is still tricky, but I think I will deal with it later in #19846.

Reply:
Thanks for the note - I think it really helps articulate what's going on.